File: 30 Days of Code/Day 12 - Inheritance/Student.java
class Student extends Person{
	private int[] testScores;

    /*	
    *   Class Constructor
    *   
    *   @param firstName - A string denoting the Person's first name.
    *   @param lastName - A string denoting the Person's last name.
    *   @param id - An integer denoting the Person's ID number.
    *   @param scores - An array of integers denoting the Person's test scores.
    */
    // Write your constructor here
    public Student(String firstName, String lastName, int id, int[] scores){
        super(firstName, lastName, id);
        this.testScores = scores;
    }
    /*	
    *   Method Name: calculate
    *   @return A character denoting the grade.
    */
    // Write your method here
    public char calculate(){
        // Get average
        int sum = 0;
        int n = testScores.length;
        for(int i=0; i< n; i++){
            sum += testScores[i];
        }
        double average = sum/n;
        
        if(average < 40){
            return 'T';
        }
        else if(average < 55){
            return 'D';
        }
        else if(average < 70){
            return 'P';
        }
        else if(average < 80){
            return 'A';
        }
        else if(average < 90){
            return 'E';
        }
        else{
            return 'O';
        }
        
        
    }
}

File: 30 Days of Code/Day 12 - Inheritance/Student.py
class Student(Person):
    #   Class Constructor
    #   
    #   Parameters:
    #   firstName - A string denoting the Person's first name.
    #   lastName - A string denoting the Person's last name.
    #   id - An integer denoting the Person's ID number.
    #   scores - An array of integers denoting the Person's test scores.
    #
    # Write your constructor here
    def __init__(self, firstName, lastName, idNum, scores):
        Person.__init__(self, firstName, lastName, idNum)
        self.scores = scores

    #   Function Name: calculate
    #   Return: A character denoting the grade.
    #
    # Write your function here
    
    def calculate(self):
        n = len(scores)
        sum = 0;
        for i in range(n):
            sum += scores[i]
        average = sum/n
        
        if(average<40):
            return 'T'
        elif(average<55):
            return 'D'
        elif(average<70):
            return 'P'
        elif(average<80):
            return 'A'
        elif(average<90):
            return 'E'
        else:
            return 'O'